tfe/resource_tfe_workspace.go

```diff
@@ -671,15 +676,54 @@ func resourceTFEWorkspaceDelete(d *schema.ResourceData, meta interface{}) error
 	id := d.Id()
 
 	log.Printf("[DEBUG] Delete workspace %s", id)
-	err := tfeClient.Workspaces.DeleteByID(ctx, id)
+
+	ws, err := tfeClient.Workspaces.ReadByID(ctx, id)
 	if err != nil {
 		if err == tfe.ErrResourceNotFound {
 			return nil
 		}
 		return fmt.Errorf(
-			"Error deleting workspace %s: %w", id, err)
+			"Error reading workspace %s: %w", id, err)
+	}
+
+	forceDelete := d.Get("force_delete").(bool)
+
+	// presence of Permissions.CanForceDelete will determine if current version of TFE supports safe deletes
+	if ws.Permissions.CanForceDelete == nil {
+		if forceDelete {
+			err = tfeClient.Workspaces.DeleteByID(ctx, id)
+		} else {
+			return fmt.Errorf(
+				"Error deleting workspace %s: This workspace must be force deleted by setting force_delete=true", id)
+		}
+	} else if *ws.Permissions.CanForceDelete {
+		if forceDelete {
+			err = tfeClient.Workspaces.DeleteByID(ctx, id)
+		} else {
+			err = tfeClient.Workspaces.SafeDeleteByID(ctx, id)
+			if err != nil {
+				if strings.Contains(err.Error(), "conflict") {
+					return fmt.Errorf(
+						"Error deleting workspace %s: %w\nTo delete this workspace without destroying the managed resources, add force_delete=true to the resource config", id, err)
+				}
+			}
+		}
+	} else {
+		if forceDelete {
+			return fmt.Errorf(
+				"Error deleting workspace %s: missing required permissions to set force_delete=true", id)
+		} else {
+			err = tfeClient.Workspaces.SafeDeleteByID(ctx, id)
```

Comment:
I was about to comment that I thought this error needs the same handling [as the one here](https://github.com/hashicorp/terraform-provider-tfe/pull/675/files#diff-e727ae216721eb3c1f24911e8b1ececae3ae9208f19b886eb627c76236471e13R703), but I caught myself at the last second 😂 

Holy %$#% this stuff is complex, major props for untangling it all! I couldn't find a single flaw with any of these cases, even after a triple check 🙇